lib/AST/ASTScalarMutationMatcher.cpp

```diff
@@ -0,0 +1,61 @@
+#include "mull/AST/ASTScalarMutationMatcher.h"
+
+#include "mull/Program/Program.h"
+
+#include <clang/AST/RecursiveASTVisitor.h>
+#include <clang/Basic/SourceLocation.h>
+
+using namespace mull;
+
+static clang::SourceLocation ClangCompatibilityStmtGetBeginLoc(const clang::Stmt &stmt) {
+#if LLVM_VERSION_MAJOR >= 7
+  return stmt.getBeginLoc();
+#else
+  return stmt.getLocStart();
+#endif
+}
+
+static bool isConstant(const clang::Stmt &statement) {
+  if (clang::isa<clang::IntegerLiteral>(&statement)) {
+    return true;
+  }
+  return false;
+}
+
+ASTScalarMutationMatcher::ASTScalarMutationMatcher(clang::ASTContext &context)
+    : context(context) {}
+
+bool ASTScalarMutationMatcher::isMutableExpr(const clang::Expr &expr,
+                                             const clang::Stmt **mutableParentStmt,
+                                             clang::SourceLocation *mutationLocation) {
+  std::cout << "hello\n";
```

Comment:
👋

Reply:
The true story behind this :)

I was creating this class from ASTVisitor that had `diagnostics` as class member. At some point I got compiler warnings that that field was not used. I decided to add this hello to just make compiler happy. Then I realized that I no longer need diagnostics there but I still decided to keep this placeholder for debugging. And then I forgot about it. It is not that simple :)